### ISSUE TITLE
Layering: Support?

### ISSUE THREAD

**Issue by saptarshiguha** [hrbrmstr/vegalite#31]
Hello,
Thanks much for the great package (vegalite is lovely). Do you have plans to support (or have i missed it?) layering (see [https://vega.github.io/vega-lite/docs/layer.html](https://vega.github.io/vega-lite/docs/layer.html) )

**Comment by saptarshiguha** [comment]
For example, i forked the repo and added this

``
mark_rule <- function(vl, ...) {
  vl <- mark(vl, mark="rule", ...)
  return(vl)
}
``

and this code does indeed replicate the example in the above link

```
p1 <- vegalite(viewport_width=800, viewport_height=600) %>%
    view_size(700,500)
p2 <-add_data(p1,"https://vega.github.io/vega-editor/app/data/stocks.csv") %>%
    encode_x("date", "temporal") %>%
    encode_y("price", "quantitative") %>%
    encode_color("symbol","nominal") %>%
    mark_line()

p3 <-add_data(p1,"https://vega.github.io/vega-editor/app/data/stocks.csv") %>%
    encode_y("price", "quantitative",aggregate='mean') %>%
    encode_size(value=2) %>%
    encode_color("symbol", "nominal") %>%
    mark_rule()

from_spec(
    sprintf("{\"layer\":[%s]}",
        paste(to_spec(p2)[[1]],
              to_spec(p3)[[1]],sep=",",collapse="")))
```

**Comment by jsonbecker** [comment]
With vegalite 2.0 now in, it's conceptually possible. But we have struggled with what the layer API should look like as you stated.

**Comment by jsonbecker** [comment]
See also #16

**Comment by jsonbecker** [comment]
See also #16

**Comment by saptarshiguha** [comment]
Yes, API design is difficult. As a stop gap, is there a way to return composable JSON that i could combine using list statements and then recombine into vega using from_spec?

The above code with sprintf and constructing JSON strings is a bit kludgy. I'm thinking i could play with layers and with the experience imagine an API that might suit my workflow.

Maybe there already is an elegant way using the current rvegalite api to do the above.

thanks

**Comment by jsonbecker** [comment]
The vegalite object passed around in these functions is a list that is designed to tranlaste to the JSON object. Check out `to_spec` to see.

**Comment by saptarshiguha** [comment]
I'm commenting here, though it's related to another issue (and this too).

```
library(jsonlite)
u <- from_spec( toJSON(. fromJSON(to_spec(u) ) ) )
u
```

plots nothing. but

```
library(jsonlite)
u <- from_spec( to_spec(u) )
u
```
works. I do need the `fromJSON` bit so i can modify the produced JSON to add bits (e.g. [https://vega.github.io/vega-lite/docs/title.html](titles)) to the spec and then plot again.

Is there way to do this using rjsonlite's JSON conversion?
